state/execution.go

```diff
@@ -381,6 +342,56 @@ func execBlockOnProxyApp(
 	return abciResponses, nil
 }
 
+func getCreateBlockValidatorInfo(
```

Comment:
add a TODO saying that in the future we may want to combine this part of code with `BeginBlock`